webpack.config.js

```diff
@@ -29,51 +29,38 @@ module.exports = {
   },
   resolve: {
     extensions: ['.webpack.js', '.web.js', '.ts', '.tsx', '.js'],
-  },
-  node: {
-    fs: 'empty',
-    tls: 'empty'
+    fallback: {
+      fs: false,
+      tls: false,
+    },
   },
   module: {
     rules: [
       {
         test: /\.(tsx|ts)?$/,
+        use: 'ts-loader',
         exclude: /node_modules/,
-        loader: 'ts-loader',
-      },
-      {
-        enforce: 'pre',
-        test: /\.js$/,
-        exclude: /node_modules/,
-        loader: 'source-map-loader',
       },
     ],
   },
   devServer: {
-    historyApiFallback: true,
+    historyApiFallback: {
+      index: `/${app}.html`,
+    },
     proxy: {
-      '/': {
```

Comment:
QQ: if we remove `/` from proxy here and same from the server.js which returns the `index.html`, does that mean now only access the app using `0.0.0.0:9000`?

Reply:
Yes, that's right. Now any request other than `'/join', '/attendee', '/meeting', '/logs', '/end'` is handled by `webpack-dev-server` via the option `historyApiFallback`:
```
historyApiFallback: {
  index: `/${app}.html`,
},
```
This will return the home page of the demo App when receiving those requests.

Reply:
So when webpack detects the requests are within `['/join', '/attendee', '/meeting', '/logs', '/end']`, it redirect them to our `server.js`. For the rest of the requests, there are no specific route for them. But with `historyApiFallback` they simply falls back to `/${app}.html` when the route is not found

Reply:
https://stackoverflow.com/questions/31945763/how-to-tell-webpack-dev-server-to-serve-index-html-for-any-route